docs/static/dead-letter-queues.asciidoc

```diff
@@ -0,0 +1,149 @@
+[[dead-letter-queues]]
+=== Dead Letter Queues
+
+NOTE: The dead letter queue feature is currently supported for the
+<<plugins-outputs-elasticsearch>> output only. Support for additional outputs
+will be available in future releases of the Logstash plugins. Before configuring
+Logstash to use this feature, refer to the output plugin documentation to
+verify that the plugin supports the dead letter queue feature.
+
+By default, when Logstash encounters an event that it cannot process because the
+data contains a mapping error or some other issue, the Logstash pipeline 
+either hangs or drops the unsuccessful event. In order to protect against data
+loss in this situation, you can configure Logstash to write unsuccessful events
+to a dead letter queue instead of dropping them. 
+
+Each event written to the dead letter queue includes the original event along
+with metadata indicating when the event entered the queue. For example:
+
+//TODO: Need a better example here. 
+
+[source,json]
+-------------------------------------------------------------------------------
```

Comment:
This is an event that I pulled out of the DLQ created by the ES plugin:
`
{
    "@timestamp" => 2017-06-16T20:47:02.510Z,
     "@metadata" => {
        "dead_letter_queue" => {
              "plugin_id" => "095bc531617b5a5fe5f632e85c9a5a03ddb204a5-2",
                 "reason" => "Could not index event to Elasticsearch. status: 404, action: [\"update\", {:_id=>\"1\", :_index=>\"logstash-2017.06.16\", :_type=>\"logs\", :_routing=>nil, :_retry_on_conflict=>1}, #<LogStash::Event:0x31ac3ede>], response: {\"update\"=>{\"_index\"=>\"logstash-2017.06.16\", \"_type\"=>\"logs\", \"_id\"=>\"1\", \"status\"=>404, \"error\"=>{\"type\"=>\"document_missing_exception\", \"reason\"=>\"[logs][1]: document missing\", \"index_uuid\"=>\"FHQaa_FKRcyIB6ef8t9qFQ\", \"shard\"=>\"3\", \"index\"=>\"logstash-2017.06.16\"}}}",
             "entry_time" => #<Java::OrgLogstash::Timestamp:0x7e5e7c82>,
            "plugin_type" => "elasticsearch"
        }
    },
      "@version" => "1",
          "host" => "dev.local",
       "message" => "Update"
}
`